test/manifest.e2e-spec.ts

```diff
@@ -386,8 +419,354 @@ describe('ganache e2e tests', () => {
+        withdrawalCredentials: fromHexString(BAD_WC),
+        amount: 1000000000, // gwei!
+      };
+
+      // Weird sig
+      const weirdDepositMessage = {
+        pubkey: pk,
+        withdrawalCredentials: fromHexString(BAD_WC),
+        amount: 0, // gwei!
+      };
+      const weirdSigningRoot = computeRoot(weirdDepositMessage);
+      const weirdSig = sk.sign(weirdSigningRoot).toBytes();
+
+      const badDepositData = {
+        ...badDepositMessage,
+        signature: weirdSig,
+      };
+      const badDepositDataRoot = DepositData.hashTreeRoot(badDepositData);
+
+      if (!process.env.WALLET_PRIVATE_KEY) {
+        process.exit();
+      }
+      const wallet = new ethers.Wallet(process.env.WALLET_PRIVATE_KEY);
+
+      // Make a bad deposit
+      const signer = wallet.connect(providerService.provider);
+      const depositContract = DepositAbi__factory.connect(
+        DEPOSIT_CONTRACT,
+        signer,
+      );
+      await depositContract.deposit(
+        badDepositData.pubkey,
+        badDepositData.withdrawalCredentials,
+        badDepositData.signature,
+        badDepositDataRoot,
+        { value: ethers.constants.WeiPerEther.mul(1) },
+      );
+
+      // Mock Keys API again on new block
+      const newBlock = await providerService.provider.getBlock('latest');
+      mockKeysApi([goodSig], newBlock, keysApiService);
+
+      // Run a cycle and wait for possible changes
+      await guardianService.handleNewBlock();
+      await new Promise((res) => setTimeout(res, SLEEP_FOR_RESULT));
+
+      // Check if on pause now
+      const routerContract = StakingRouterAbi__factory.connect(
+        STAKING_ROUTER,
+        providerService.provider,
+      );
+      const isOnPause = await routerContract.getStakingModuleIsDepositsPaused(
+        1,
+      );
+      expect(isOnPause).toBe(false);
+    },
+    TESTS_TIMEOUT,
+  );
+
+  test(
+    'good scenario',
+    async () => {
+      const tempProvider = new ethers.providers.JsonRpcProvider(
+        `http://127.0.0.1:${GANACHE_PORT}`,
+      );
+      const currentBlock = await tempProvider.getBlock('latest');
+
+      const goodDepositMessage = {
+        pubkey: pk,
+        withdrawalCredentials: fromHexString(GOOD_WC),
+        amount: 32000000000, // gwei!
+      };
+      const goodSigningRoot = computeRoot(goodDepositMessage);
+      const goodSig = sk.sign(goodSigningRoot).toBytes();
+
+      mockKeysApi([goodSig], currentBlock, keysApiService);
+
+      const goodDepositData = {
+        ...goodDepositMessage,
+        signature: goodSig,
+      };
+      const goodDepositDataRoot = DepositData.hashTreeRoot(goodDepositData);
+
+      await depositService.setCachedEvents({
+        events: [],
+        startBlock: currentBlock.number,
+        endBlock: currentBlock.number,
+      });
+
+      // Check if the service is ok and ready to go
+      await guardianService.handleNewBlock();
+
+      if (!process.env.WALLET_PRIVATE_KEY) {
+        process.exit();
```

Comment:
Maybe we should print some messages except `process.exit`?

Reply:
Yeah, changed not for all, fixing, thanks!

Reply:
Done